streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -157,26 +165,82 @@ public void shouldIdempotentlyUpdateSubscriptionFromActiveAssignment() {
     }
 
     @Test
-    public void shouldReturnOffsetsForAllCachedTaskIdsFromDirectory() throws IOException {
-        final File[] taskFolders = asList(testFolder.newFolder("0_1"),
-                                          testFolder.newFolder("0_2"),
-                                          testFolder.newFolder("0_3"),
-                                          testFolder.newFolder("1_1"),
-                                          testFolder.newFolder("dummy")).toArray(new File[0]);
+    public void shouldReportOffsetSumsForValidLockedTasks() throws IOException {
```

Comment:
req: In general I think this unit test is really large. For the sake of readability and modularization, you should split it into multiple tests. Maybe for each case two unit tests: one with a single case. Then one unit test for the composite scenario with all cases and different occurrences of the different cases. If you extract and parametrize the setup, it should not be too much code duplication. Additionally, a test where `stateDirectory.listTaskDirectories()` returns an empty array and a test with a stateless task are missing.

Reply:
To clarify, you're suggesting to add smaller tests for each case (and edge cases) but also leave the composite test in as well?

Reply:
Basically yes. I would extend the composite test with multiple occurrences of each case so that we also cover the scenario where we have -- for example -- n active running task, m active non-running tasks, k standby tasks etc. 
If this makes the test too clumsy, you could cover each of n active running task, m active non-running tasks, k standby tasks etc in its own test and make one composite test with one occurrence of each case. Choose what is better readable.
My point is, that the current test does not cover multiple occurrences of the same case.